File: projetJeuMaven/src/main/java/fr/projetjeu/model/Personnage.java
package fr.projetjeu.model;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
@Table(name = "personnage")
public class Personnage {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY) // Auto-increment
	@Column(name = "per_id")
	private int id;
	
	@Column(name = "per_nom", length=50, nullable = false)
	private String nom;
	
	@Column(name = "per_prenom", length=50, nullable = false)
	private String prenom;

	@Column(name = "per_poids", nullable = false)
	private float poids  =0;
	
	@Column(name = "per_argent", nullable = false)
	private float argent=0;
	
	@Column(name = "per_energie", nullable = false)
	private float energie=0;
	
	@Column(name = "per_faim", nullable = false)
	private float faim=0;
	
	@Column(name = "per_force", nullable = false)
	private float force=0;
	
	@Column(name = "per_covid", nullable = false )
	private boolean isCovided=false;
	
	@Column(name = "per_vivant", nullable = false)
	private boolean isAlive=true;

	@Enumerated(EnumType.ORDINAL)
	@Column(name="per_humeur",nullable=false)
	private Humeur humeur;
	

	@ManyToMany
	@JoinTable(
			name = "competence_personnage",
			joinColumns = @JoinColumn(name  ="comper_personnage_id"),
			inverseJoinColumns = @JoinColumn(name = "comper_competence_id")
		)
	private List<Competence> competences;

	
	@OneToOne(mappedBy  ="personnage")
	private Partie partie;
	

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	public float getPoids() {
		return poids;
	}

	public void setPoids(float poids) {
		this.poids = poids;
	}

	public float getArgent() {
		return argent;
	}

	public void setArgent(float argent) {
		this.argent = argent;
	}

	public float getEnergie() {
		return energie;
	}

	public void setEnergie(float energie) {
		this.energie = energie;
	}

	public float getFaim() {
		return faim;
	}

	public void setFaim(float faim) {
		this.faim = faim;
	}

	public float getForce() {
		return force;
	}

	public void setForce(float force) {
		this.force = force;
	}

	public boolean isCovided() {
		return isCovided;
	}

	public void setCovided(boolean isCovided) {
		this.isCovided = isCovided;
	}

	public Humeur getHumeur() {
		return humeur;
	}

	public void setHumeur(Humeur humeur) {
		this.humeur = humeur;
	}



	public List<Competence> getCompetences() {
		return competences;
	}

	public void setCompetences(List<Competence> competences) {
		this.competences = competences;
	}

	public Partie getPartie() {
		return partie;
	}

	public void setPartie(Partie partie) {
		this.partie = partie;
	}

	public boolean isAlive() {
		return isAlive;
	}

	public void setAlive(boolean isAlive) {
		this.isAlive = isAlive;
	}

	
}

File: ProjetAngular/src/app/pagesweb/model/personnage.ts
import { Competence } from "./competence";

export class Personnage {
  public get id(): number |undefined{
    return this._id;
  }
  public set id(value: number |undefined) {
    this._id = value;
  }
  public get competences(): Competence[] |undefined{
    return this._competences;
  }
  public set competences(value: Competence[]|undefined) {
    this._competences = value;
  }
  public get humeur(): string |undefined{
    return this._humeur;
  }
  public set humeur(value: string|undefined) {
    this._humeur = value;
  }
  public get isAlive(): boolean|undefined {
    return this._isAlive;
  }
  public set isAlive(value: boolean|undefined) {
    this._isAlive = value;
  }
  public get isCovided(): boolean|undefined{
    return this._isCovided;
  }
  public set isCovided(value: boolean|undefined) {
    this._isCovided = value;
  }
  public get force(): number|undefined {
    return this._force;
  }
  public set force(value: number|undefined) {
    this._force = value;
  }
  public get faim(): number|undefined {
    return this._faim;
  }
  public set faim(value: number|undefined) {
    this._faim = value;
  }
  public get energie(): number|undefined{
    return this._energie;
  }
  public set energie(value: number|undefined) {
    this._energie = value;
  }
  public get argent(): number |undefined{
    return this._argent;
  }
  public set argent(value: number|undefined) {
    this._argent = value;
  }
  public get poids(): number |undefined{
    return this._poids;
  }
  public set poids(value: number|undefined) {
    this._poids = value;
  }
  public get prenom(): string |undefined{
    return this._prenom;
  }
  public set prenom(value: string|undefined) {
    this._prenom = value;
  }
  public get nom(): string|undefined {
    return this._nom;
  }
  public set nom(value: string|undefined) {
    this._nom = value;
  }

  constructor (private _id?: number,
              private _nom?: string,
              private _prenom?: string,
              private _poids?: number,
              private _argent?: number,
              private _energie?: number,
              private _faim?: number,
              private _force?: number,
              private _isCovided?: boolean,
              private _isAlive?: boolean,
              private _humeur?: string,
              private _competences?: Competence[]
              ){}
}
